opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -409,7 +410,19 @@ static inline int mca_btl_base_am_rdma_advance(mca_btl_base_module_t *btl,
     }
 
     if (send_descriptor) {
-        return btl->btl_send(btl, endpoint, descriptor, mca_btl_base_rdma_tag(hdr->type));
+        /* btl_send return OPAL_SUCCESS when that the descriptor has been queued for send, and
```

Comment:
You don't need this big of a comment for the semantics (now that the btl.h file has the documentation on return value).  I would add a note that we always set `MCA_BTL_DES_SEND_ALWAYS_CALLBACK`, so we can treat 0 and 1 the same.  Just something as simple as:

```
/* This descriptor will have MCA_BTL_DES_SEND_ALWAYS_CALLBACK set, so safe to treat 0 and 1 return codes the same */
```

Or you could replace that comment with the even better assert:
```
assert(0 != (descriptor->flags && MCA_BTL_DES_SEND_ALWAYS_CALLBACK));
```
that way, you've both documented the assumption and proven the assumption (in debug builds, anyway).

Reply:
Thanks! In new revision, I removed the excessive comments, and added the assertion.